src/safe-event-emitter-provider.test.ts

```diff
@@ -0,0 +1,119 @@
+import { JsonRpcEngine } from 'json-rpc-engine';
+import { promisify } from 'util';
+
+import { SafeEventEmitterProvider } from './safe-event-emitter-provider';
+
+describe('SafeEventEmitterProvider', () => {
+  describe('constructor', () => {
+    it('emit engine notifications from provider', async () => {
```

Comment:
This makes it sound like the constructor immediately emits the notifications. Perhaps something like this?

``` suggestion
    it('listens for notifications from provider, emitting them as "data"', async () => {
```

Reply:
Good point!

Reply:
Addressed in https://github.com/MetaMask/eth-json-rpc-provider/pull/6/commits/2ac2b8063d30dffdf68473a3f66c82b49a178958